pkg/local_object_storage/writecache/writecache.go

```diff
@@ -143,8 +144,8 @@ func (c *cache) Open(readOnly bool) error {
 }
 
 // Init runs necessary services.
-func (c *cache) Init() error {
-	c.initFlushMarks()
+func (c *cache) Init(async bool) error {
```

Comment:
yes, that's an ugly change. problems appear when you try to (unit)test it. `WriteCache` should either be able to init in sync mode or have a feedback channel to signal about its ready state then. i decided to open PR anyway and discuss that. your thoughts?

Reply:
vote for new option for writecache `WithAsyncInitialization` and init it in background by default.

Reply:
Can we just use `require.Eventually` in tests?

Reply:
> Can we just use require.Eventually in tests?

yes, but with magic numbers (how oftern and how long wait for that status (and how? try to `Put` object? export `Status` func?)). do we want it?

> vote for new option for writecache WithAsyncInitialization and init it in background by default.

we have `Init` -> `Close` -> `Init` scenario. i could imagine that sync/async initialization could be a config value in future. also, IMO creation option should not change `Init` behavior

Reply:
> do we want it?
We already have it in some places, I don't mind having it here.

> i could imagine that sync/async initialization could be a config value in future
Well, it could, but we don't _need_ it now.

> creation option should not change Init behavior
Why so?

Reply:
> I don't mind having it here

ok, dropped `async` flag